lib/Service/GroupsIntegration.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+/**
+ * @author Matthias Rella <mrella@pisys.eu>
+ *
+ * Mail
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\Mail\Service;
+
+use OCA\Mail\Service\Group\AbstractGroupService;
+
+class GroupsIntegration {
+
+	/**
+	 * The services to get groups from
+	 *
+	 * @var IGroupService[]
+	 */
+	private $groupServices = [];
+
+	public function __construct(AbstractGroupService ...$groupServices) {
```

Comment:
For simplicity, I'd recommend to only inject a single group service (there's only one so far). You don't have to use the concrete impl here but use the interface (as proposed above) and register the impl as alias for the interface in `Application`.

Reply:
I designed this so it will be extensible for Contacts groups. 
But I could reserve this feature for later when Contacts groups integration will be ready.